### ISSUE TITLE
Updating the value of "catalog/seo/category_url_suffix" or "catalog/seo/product_url_suffix" can cause Magento to silently fail to update URL rewrites

### ISSUE THREAD

**Issue by simonworkhouse** [magento/magento2#17614]
After updating the value of either "catalog/seo/category_url_suffix" or "catalog/seo/product_url_suffix" it is possible to create another product or category that matches the same expected URL rewrite path as conflicts won't be reported due to the difference in URL keys.
If a common parent category of these conflicting entities has it's URL key changed, both entities may have their URL rewrites updated to have the same request path which will cause only the latest category to have the rewrites updated and no errors generated.

### Preconditions

1. Magento 2.2.5
2. PHP 7.1.20
3. MySQL 5.7.23

### Steps to reproduce

1. composer create-project --repository-url=https://repo.magento.com/ magento/project-community-edition {install dir}
2. cd {install dir}
3. ./bin/magento setup:install --backend-frontname=? --db-host=? --db-name=? --db-user=? --db-password=? --base-url=? --admin-user=? --admin-password=? --admin-email=? --admin-firstname=? --admin-lastname=?
4. ./bin/magento deploy:mode:set developer
5. ./bin/magento app:config:dump
6. ./bin/magento cache:disable
7. Confirm that the value for "catalog/seo/category_url_suffix" is ".html" (the default value)
8. Proceed to the admin panel and create a sub-category called "Sale" with the URL key of "sale"
9. Create a sub-category under "Sale" called "Sandwiches" with the URL key of "sandwiches"
10. Now update the name of the new category from "Sandwiches" to "Sandwiches Old" (NOTE: This step is just to simulate user behaviour and highlight the potential of conflicts when importing data)
11. Update the value of "catalog/seo/category_url_suffix" to ".php" and run "./bin/magento app:config:import"
12. Create another sub-category under "Sale" called "Sandwiches" with the URL key of "sandwiches"
13. Update the value of "catalog/seo/category_url_suffix" to ".test" and run "./bin/magento app:config:import"
14. Update the URL key for the "Sale" category from "sale" to "sale-test"

### Expected result

The URL keys for categories should never have been allowed to conflict like this and at the very least a warning notifying the admin user of conflicts in the url_rewrite table should be shown.

### Actual result

A success message saying "You saved the category." is displayed to the admin user and only the latest category/product has the URL rewrites updated.

**Comment by magento-engcom-team** [comment]
Hi @simonworkhouse. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation.

@simonworkhouse do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by engcom-backlog-nickolas** [comment]
Hello @simonworkhouse, thank you for your report. I'm not able to reproduce this issue. On step 12:

>  Create another sub-category under "Sale" called "Sandwiches" with the URL key of "sandwiches"

I've got error message:
![url_rewrite](https://user-images.githubusercontent.com/40755810/44134768-f6e12158-a06e-11e8-93fa-a2e0eda93c7a.png)

Could you please add more details?

**Comment by simonworkhouse** [comment]
@engcom-backlog-nickolas Is your cache disabled? Try a "./bin/magento cache:flush" as the final URL should end in ".test" not ".php"

**Comment by engcom-backlog-nickolas** [comment]
@simonworkhouse, yes cache is disabled. Why it should ended with ".test"? Suffix ".test" is set at step 13. And I cannot get to 13th step.

**Comment by simonworkhouse** [comment]
@engcom-backlog-nickolas Yes, try monitoring changes to the url_rewrite as you proceed through the steps. I'll have someone re-check the instructions.

**Comment by simonworkhouse** [comment]
@engcom-backlog-nickolas I have had others independently run through the process and they were able to replicate the issue as described. Also, you won't be able to test this issue and the other related issues that have been reported at the same time, you will have to start from a fresh install for each.

**Comment by engcom-backlog-nickolas** [comment]
@simonworkhouse 
![peek 2018-08-15 16-04](https://user-images.githubusercontent.com/40755810/44149751-dad9a4b2-a0a5-11e8-8ddf-7c4de9da5f51.gif)

Please correct my steps, or add more details.

**Comment by simonworkhouse** [comment]
@engcom-backlog-nickolas Try updating the url suffixes directly in the config.php file.

**Comment by simonworkhouse** [comment]
This issue is still relevant.

**Comment by simonworkhouse** [comment]
This issue is still relevant.

**Comment by engcom-Bravo** [comment]
Hi @simonworkhouse,

Thanks for your reporting and collaboration.

We have tried to reproduce the issue in Latest 2.4-develop instance and the issue is no more reproducible.Hence we are closing this issue.

Thanks.
